paper-server/patches/sources/net/minecraft/server/commands/WaypointCommand.java.patch

```diff
@@ -0,0 +1,11 @@
+--- a/net/minecraft/server/commands/WaypointCommand.java
++++ b/net/minecraft/server/commands/WaypointCommand.java
+@@ -165,7 +_,7 @@
+     }
+ 
+     private static void mutateIcon(CommandSourceStack source, WaypointTransmitter waypoint, Consumer<Waypoint.Icon> mutator) {
+-        ServerLevel level = source.getLevel();
++        ServerLevel level = (waypoint instanceof LivingEntity livingEntity) ? (net.minecraft.server.level.ServerLevel) livingEntity.level() : source.getLevel(); // Paper - use level of waypoint if it's a living entity for broadcast
```

Comment:
this is based in the list argument where the same check is make for the waypoints, not sure in what case the waypoint cannot be a LivingEntity